### ISSUE TITLE
Missing developer building guide

### ISSUE THREAD

**Issue by wolfch** [DapperLib/Dapper#1127]
I am trying to build for dotnet-core 2.0 and not having luck.  I tried running `build.sh`
```
$ ./build.sh

Installing dotnet cli...

./build.sh: line 9: tools/install.sh: No such file or directory
Set version to rc2-1419317-52
sed: can't read s/99.99.99-rc2/1.0.0-rc2-1419317-52/g: No such file or directory
sed: can't read */*/project.json: No such file or directory
```
Why would it try to install "dotnet cli" when I already have it installed?

Next, I tried building with `build.cmd` - that seemed to have gone further - all lot more steps completed, until the build finally failed with:
```
Build FAILED.

C:\Program Files\dotnet\sdk\2.1.400\Microsoft.Common.CurrentVersion.targets(1179,5): 
error MSB3644: The reference assemblies for framework ".NETFramework,Version=v4.6.2" were not found. 
To resolve this, install the SDK or Targeting Pack for this framework version or retarget your application to a version of the framework for which you have the SDK or Targeting Pack installed. 
Note that assemblies will be resolved from the Global Assembly Cache (GAC) and will be used in place of reference assemblies. 
Therefore your assembly may not be correctly targeted for the framework you intend. 
[C:\Users\cwolf\src\wolfch\Dapper\Dapper.Tests.Performance\Dapper.Tests.Performance.csproj]
    0 Warning(s)
    1 Error(s)
```
I have dotnet framework `v4.7` installed - will it really only build with _exactly_ `v4.6.2` ?

Bottom line:  How do we checkout and build Dapper and all its subprojects?

**Comment by NickCraver** [comment]
I'm not sure the build works in Linux at all right now - we should probably remove the `build.sh` script to remove confusion. With Appveyor we're building Windows and Linux with PowerShell in other libraries.

**Comment by wolfch** [comment]
Ok, I initially tried `build.sh` because it was there and I use bash/WSL-Ubuntu on Windows-10.  I don't care about Linux for now - I just need to know how to build so I can submit a bug fix PR.  I revised my initial post to include my attempt to build with `build.cmd` which doesn't work either.

**Comment by wolfch** [comment]
Anyway, since I have Windows10 it's _impossible_ to downgrade to dotnet framework `v4.6.2` or even install to coexist with the provided `v4.7`.
https://docs.microsoft.com/en-us/dotnet/framework/install/troubleshoot-blocked-installations-and-uninstallations

**Comment by NickCraver** [comment]
@wolfch To target .NET 4.6.2 (which we also do), it needs DLLs to compile against (called reference assemblies). You can install the .NET 4.6.2 targeting pack via Visual Studio (or just the build tools)'s installer or a quick search + install. Later this will improve as the .NET team is making that story work via NuGet, but for now...that's where we are.

You don't need to change your version of .NET, just install the reference assemblies for any versions you want to compile against.

**Comment by wolfch** [comment]
@NickCraver - thanks so much for the help! I installed the "targeting pack" from this offline installer:
https://www.microsoft.com/en-us/download/details.aspx?id=53321
Then I re-attempted `build.cmd` and this time the build succeeded!  I'm mostly a Java guy - so this NuGet stuff is still nu to me!

**Comment by wolfch** [comment]
I still recommend an addendum to the readme for developer building.

**Comment by wolfch** [comment]
I spoke too soon - the build succeeded:
```
Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:01:56.26
Done building.
```
...but most of the tests fail.
```
[...]
Failed   Dapper.Tests.MiscTests.TestDapperTableMetadataRetrieval
Error Message:
 System.Data.SqlClient.SqlException : A network-related or instance-specific error occurred while establishing a connection to SQL Server. 
The server was not found or was not accessible. Verify that the instance name is correct 
and that SQL Server is configured to allow remote connections. 
(provider: Named Pipes Provider, error: 40 - Could not open a connection to SQL Server)
[...]
Total tests: 247. Passed: 9. Failed: 225. Skipped: 13.
Test Run Failed.
Test execution time: 1.6569 Minutes
Error with tests, aborting build.
```

Seems like there's some undocumented SQL Server setup required.  I wonder what other undocumented setups are required for the tests to run?

**Comment by NickCraver** [comment]
@wolfch Yes you'll need SQL Server installed, but to run the full suite there's also MySQL, Postgres, Firebase, Sqlite, etc. (later this will include Oracle as well).

In short: Dapper is a thing that communicates with a database, the test setup for a client library is never super simple due to the basic requirements. But we don't intend for it to be either - our PR builds run all these tests and checks for you. The goal isn't to make setup for the full test suite easier, it's to make it *not needed at all*.

**Comment by wolfch** [comment]
Is there a way to only test SQL Server & SQLite and skip the other DB flavor tests?  (I assume SQLite would be in-memory and not require setup)

**Comment by wolfch** [comment]
Also build only works for `Release` not `Debug`.
